packages/react-workers/src/utils/sorting.js

```diff
@@ -0,0 +1,66 @@
+import { firstBy } from 'thenby';
+
+/**
+ * Apply sort structure to a collection of features
+ * @param {array} features
+ * @param {object} [sortOptions]
+ * @param {string | string[] | object[]} [sortOptions.sortBy] - One or more columns to sort by
+ * @param {string} [sortOptions.sortByDirection] - Direction by the columns will be sorted
+ */
+export function applySorting(features, { sortBy = [], sortByDirection = 'asc' } = {}) {
+  const isValidSortBy =
+    (Array.isArray(sortBy) && sortBy.length) || // sortBy can be an array of columns
+    typeof sortBy === 'string'; // or just one column
+
+  if (!isValidSortBy) {
+    return features;
+  }
+
+  const sortFn = createSortFn({
+    sortBy,
+    sortByDirection
+  });
+
+  return features.sort(sortFn);
+}
+
+// Aux
+function createSortFn({ sortBy, sortByDirection }) {
+  const [firstSortOption, ...othersSortOptions] = normalizeSortByOptions({
+    sortBy,
+    sortByDirection
+  });
+
+  let sortFn = firstBy(...firstSortOption);
+  for (let sortOptions of othersSortOptions) {
+    sortFn = sortFn.thenBy(...sortOptions);
+  }
+  
+  return sortFn;
+}
+
+function normalizeSortByOptions({ sortBy, sortByDirection }) {
+  if (!Array.isArray(sortBy)) {
+    sortBy = [sortBy];
+  }
+
+  return sortBy.map((sortByEl) => {
+    // sortByEl is 'column'
+    if (typeof sortByEl === 'string') {
+      return [sortByEl, sortByDirection];
+    }
+
+    if (Array.isArray(sortByEl)) {
+      // sortBy is ['column']
+      if (sortByEl[1] === undefined) {
+        return [sortByEl, sortByDirection];
+      }
+
+      // sortBy is ['column', { ... }]
+      if (typeof sortByEl[1] === 'object') {
+        return [sortByEl[0], { direction: sortByDirection, ...sortByEl[1] }];
+      }
+    }
+    return sortByEl;
+  });
```

Comment:
This is the part I was talking about before. Do we need all those variations regarding sortByEl structure?

Reply:
I think that if we want to offer a wide range of possibilities from the simplest to the most complex, the answer is yes. And I think that this kind of function that normalize options should be very common in C4R because these functions concentrate the complexity of making a functionality advanced and makes the rest of the code easy to read. Also, if the function is well documented (like I think this one is), this can be understood by anyone and won't be difficult to maintain. In the BarWidgetUI, I also use another normalize fn because it makes everything easier when you're dealing with different kind of inputs, one from the basic users (I just want to order by one column, I send you a string) and another for the most advanced users (I want to order by multiple columns and each one in a different way). We can offer all these variation just making this 20 lines fn without compromising the rest of the code because we're abstracting those line from that using this normalize fn.
I hope this helps to understand my vision.

Reply:
If you really need that flexibility, then I'm ok with it. I just want to be sure it's needed